code/crates/engine/src/consensus.rs

```diff
@@ -1084,6 +1094,10 @@ where
             }
 
             Effect::ExtendVote(height, round, value_id, r) => {
+                if !is_active_validator {
+                    return Ok(r.resume_with(None));
```

Comment:
Nothing to say about the logic in here. I understand that, if the core logic no longer distinguishes between validator and non-validator, it's gonna try to create a vote in all cases, hence we're asked here to extend a (useless) vote... and I understand we'll then get called to sign that vote..

I don't know if you have thought of introducing an effect, say `ShouldVote` which returns a boolean. So that consensus doesn't waste resources in creating, extending, and signing votes that -- conceptually -- make sense as votes with power 0 (so useless in practice).